ui/src/app/pages/analysis/notebook-redirect.tsx

```diff
@@ -251,58 +272,86 @@ export const NotebookRedirect = fp.flow(withUserProfile(), withCurrentWorkspace(
       return this.props.queryParams.playgroundMode === 'true';
     }
 
-    private async pollCluster(billingProjectId) {
-      const repoll = () => {
-        this.clearAndSetTimeout(() => this.pollCluster(billingProjectId), 15000);
-      };
-      const {workspace} = this.props;
-      const {initialized} = this.state;
+    private async initializeClusterStatusChecking(billingProjectId) {
       try {
-        const resp = await clusterApi().listClusters(billingProjectId, workspace.id,
-          {signal: this.aborter.signal});
-        const cluster = resp.defaultCluster;
-        if (!initialized) {
-          if (cluster.status === ClusterStatus.Running) {
-            this.incrementProgress(Progress.Unknown);
-          } else if (this.isClusterInProgress(cluster)) {
+        const cluster = await this.getDefaultCluster(billingProjectId);
+
+        if (cluster.status === ClusterStatus.Running) {
+          this.incrementProgress(Progress.Unknown);
+          await this.connectToRunningCluster(cluster);
+        } else {
+          if (this.isClusterInProgress(cluster)) {
             this.incrementProgress(Progress.Resuming);
           } else {
             this.incrementProgress(Progress.Initializing);
           }
-          this.setState({initialized: true});
+
+          this.scheduleClusterPoll(billingProjectId);
```

Comment:
`initializeClusterStatusChecking()` is separated out because I think that makes more sense than a single-use "initialized" state variable that requires a check each time.  `connectToRunningCluster()` is split out because there are two paths to get there: via the initial status check on an already-running cluster, and and via polling.

I reordered the methods to make this hopefully easier to follow.